src/components/table/table.js

```diff
@@ -358,10 +358,8 @@ export default {
           role: this.isStacked ? 'table' : null,
           'aria-busy': this.computedBusy ? 'true' : 'false',
           'aria-colcount': String(fields.length),
-          'aria-rowcount':
-            this.$attrs['aria-rowcount'] || (this.perPage && this.perPage > 0)
-              ? '-1'
-              : null
+          'aria-rowcount': this.$attrs['aria-rowcount'] ||
+            this.items.length > this.perPage ? this.items.length : null
```

Comment:
This assumes that `this.items` is an array.  It isn't when it is a provider function. When a provider, `this.items.length` actually returns the argument count the provider function is expecting (i.e. 1 or 2).

PR #2149 fixes this.